python/private/python.bzl

```diff
@@ -850,6 +857,19 @@ The coverage tool to be used for a particular Python interpreter. This can overr
         ),
     },
 )
+_converage = tag_class(
```

Comment:
For now this toolchain is completely separate from the `python` toolchain and it would be impossible to extend it in the future to be able to use dependencies from PyPI, because `python` extension is used inside the `pip` extension.

For this reason, I would like us to consider adding a separate `bzlmod` extension, especially, since this maybe could be always run with `dev_dependency` as it is about the test env.

In general this could have an API:
```
python_test = use_extension("@rules_python//python/extensions:python_test.bzl", "python_test")

python_test.configure(
    testrunner = "my_label_for_test_tool",  # could be `pytest-bazel` PyPI dep
    coverage_tool = "my_label_for_coverage_tool",  # should default to the rules_python provided one
    coveragerc = ".coveragerc",
)
```

The users who want a completely custom experience could define their own `pytest_toolchain` and register it instead of using the one defined by us.